nservicebus/bridge/migrating-from-router.md

```diff
@@ -0,0 +1,82 @@
+
+Because of this, the NServiceBus.Router must remain running until all in-flight messages have been delivered. The time this takes is highly individual to each system. If it is unclear what the right time is to stop the NServiceBus.Router, contact [Particular Software](https://particular.net/contactus) for guidance.
+
+#### Commands
+
+Commands are the easiest to migrate as they have a specific route to a specific logical endpoint. After the transport bridge has been set up and running, every endpoint can remove any NServiceBus.Router-related configuration for routing commands using the connector. This implies that the original routing towards the specific endpoint can be restored and the bridge will make sure it is transferred to the correct transport.
+
+#### Events
+
+Events are the hardest message type to migrate to the transport bridge from NServiceBus.Router.
+
+The issue is with the subscriptions that are registered within a message broker. The scenario could exist as seen in the graph below, where `Endpoint B` publishes an event and the event is sent to both the `NServiceBus.Router` and the `NServiceBus.Transport.Bridge`. As a result, `Endpoint C` could receive the same event twice.
+
+Note that both events would have the same message identifier, which allows the [outbox](/nservicebus/outbox/) to de-duplicate the messages. Without the outbox, either the message handler must be [idempotent](/nservicebus/concepts/glossary.md#idempotence) or the event will be processed twice.
+
+```mermaid
+flowchart LR
+  Br(Bridge)
+  A[Endpoint A] <---> Br
+  B[Endpoint B] <---> Br
+  Br <---> C[Endpoint C]
+  Br <---> D[Endpoint D]
+  Ro[Router] <---> C
+  B <--> Ro
+  subgraph msmq
+    A
+    B
+  end
+  subgraph sql
+    C
+    D
+  end
+```
+
+For this scenario, take the following steps to move from NServiceBus.Router to NServiceBus.Transport.Bridge:
+
+1. Take down both `EndpointB` and the `Bridge`.
+1. Configure the `Bridge` to register a publisher with `Endpoint C` for the event that `Endpoint B` publishes.
+2. Remove the subscription from the message broker for the `Router`.
+3. Add the subscription to the message broker for the `Bridge`.
+4. Start the `Bridge`.
+5. Start `Endpoint B`.
+
+When the `Bridge` starts up, it will also create the subscription; it is not required to manually create the subscription in the message broker. However, tt is important to be aware of when `Endpoint B` cannot be taken offline due to high availability, scaling out or service level agreements.
```

Comment:
```suggestion
When the `Bridge` starts up, it will also create the subscription; it is not required to manually create the subscription in the message broker. However, be aware of when `Endpoint B` cannot be taken offline due to high availability, scaling out or service level agreements.
```